examples/Example3/particle.cc

```diff
@@ -0,0 +1,250 @@
+// SPDX-FileCopyrightText: 2020 CERN
+// SPDX-License-Identifier: Apache-2.0
+
+#include "particle.h"
+
+#include "user.h"
+
+#include <algorithm>
+#include <cstdio>
+#include <vector>
+
+static int ntotal;
+static int nalive;
+static std::vector<particle> particles;
```

Comment:
These are not global variables, they are only global to this file and are not visible to other parts of the code. Declaring them static or putting them in a namespace are more or less equivalent in that sense. See my comment above on namespaces.

What is currently a file in this example could probably be represented as a directory with several files later, and at least some of these statics will become class members when we have a better idea of what classes we will need. For example, in this particular case, a `ParticleSystem` class could be created to hold all tracks of a given type (like photons, electrons, etc), and the functions currently provided by this module would then be methods of the `ParticleSystem` class. However, since it's not yet clear what will be the best way to organize the code, and if more classes will be needed in the end, this module acts like a more fluid way of "the place where particles are handled" within the code. The main goal I kept for this module was not to leak into the other parts of the code how particles are stored in memory, because that's very important in order to be able to have different layouts on CPU and GPU without having to modify many other parts of the code. Since I added all of the GPU support after everything was working on the CPU already, and didn't have to change other files to do that, I think that this structure served the purpose well so far.